testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -738,7 +738,11 @@ impl BlockMinerThread {
         staging_tx.commit()?;
 
         if !accepted {
-            warn!("Did NOT accept block {} we mined", &block.block_id());
+            // this can happen if the p2p network and relayer manage to receive this block prior to
+            // the thread reaching this point -- this can happen because the signers broadcast the
+            // signed block to the nodes independent of the miner, so the miner itself can receive
+            // and store its own block outside of this thread.
+            debug!("Did NOT accept block {} we mined", &block.block_id());
```

Comment:
It seems like `NakamotoChainState::accept_block`'s `Ok` result should be an enum rather than a `bool`, e.g.:

```rust
enum BlockAcceptResponse {
   Accepted,
   AlreadyStored,
   Rejected(String)
}
```

Because the miner may really want to be able to differentiate between `Rejected` and `AlreadyStored`.

Reply:
Just to ping here -- I think its not really important that this change anything about the miner's behavior, but we should differentiate between the two "rejected" cases for the purposes of logging:

Getting a rejection on a block the miner themselves produced should generate a WARN log (or maybe even an ERROR log). This is a failure scenario that indicates something bad is happening. Whereas the block being "already stored" is not a problem really.

Reply:
Yes; I'll do this.